### DISCUSSION TITLE
Reheat Rankine Cycle

### DISCUSSION THREAD

**Discussion by alinuman15** [oemof/tespy#405] [General]
Hi fwitte:
Thank you for replying. The problem I have written was not a HW or so rater than a solved example chosen from a known textbook to compare the resuts with those obtained from Tespy.
The sketch of the problem is as shown below. I doubt the following connection since it is not reflect the reheater-boiler interaction. How could we tackle that. 
![Sketch](https://user-images.githubusercontent.com/25305359/229283914-1d147201-689b-4970-85ac-c87bc7061775.jpg)
#connections
c1 = Connection(CycleCloser, 'out1', HPT, 'in1', label='1')
c2 = Connection(HPT, 'out1', Reheater, 'in1', label='2')
c3 = Connection(Reheater, 'out1', LPT, 'in1', label='3')
c4 = Connection(LPT, 'out1', Condenser, 'in1', label='4')
c5 = Connection(Condenser, 'out1', Pump, 'in1', label='5')
c6 = Connection(Pump, 'out1', Boiler, 'in1', label='6')
c0 = Connection(Boiler, 'out1', CycleCloser, 'in1', label='0')

**Comment by fwitte** [comment]
@alinuman15, this sounds like some homework or something. Therefore two things:

1. I'd like to encourage you to try to build the model yourself and ask here for help, if you are struggeling with specific tasks. You can then ask, what the issue is specifically.
2. Please attach images etc. directly and do not link .rar, .zip or similar files, since such files might not always be trustworthy. And people do not need to navigate to other website, to get the information you are talking about.

**Reply by fwitte** [reply]
@alinuman15, sorry for the late reply. I did not get a notification on you updating your question. In this case it usually helps if you just post a new answer.

So to you question: You setup looks perfectly fine when the boiler and the reheater should just add whatever heat the cycle requires to provide the desired power, and unless you want to make a detailed calculation of the boiler and reheater. If you want to do that, it really depends: For a steam generator using hot flue gases as source of energy it is possible to create something. A coal fired power plant does not really work, because TESPy does not have the components to build that. Does that help you?

**Reply by alinuman15** [reply]
Thank you for replying @fwitte  and no worries about late reply.
I am quoating :
"For a steam generator using hot flue gases as source of energy it is possible to create something":  **Do you mean by something that we need to create a new class ?**

**Reply by fwitte** [reply]
No, you can just use the inbuilt components to build a waste heat steam generator, i.e. heat exchangers + drum, for example like this: https://tespy.readthedocs.io/en/main/modules/components.html#example. You do not need to use the class structure for the subsystem as in that example, you can simply build that topology into your script.

**Comment by diego-atehortua** [comment]
hello.

I am also considering a similar exercise, the problem occurs when connecting the superheater to the high pressure turbine, it happens that when simulating an error appears that says that the superheater is waiting for an output, that response is not logical because the connection goes from the high pressure turbine as output, enters the superheater, leaves the superheater and enters the low pressure turbine. 

Translated with www.DeepL.com/Translator (free version)

**Reply by fwitte** [reply]
Hi, could you provide a minimum example to reproduce your issue? Otherwise nobody will be able to give you hints, what the issue might be...
